packages/create-amplify/src/initial_project_file_generator.ts

```diff
@@ -26,10 +29,14 @@ export class InitialProjectFileGenerator {
       { recursive: true }
     );
 
-    const packageJsonContent = { type: 'module' };
+    const packageJsonContent: PackageJson = { type: 'module' };
     await this.fs.writeFile(
       path.resolve(targetDir, 'package.json'),
       JSON.stringify(packageJsonContent, null, 2)
     );
+
+    // there's a bit of tight coupling here because this class sets type: module in the package.json for the directory and tsConfigInitializer create a TS config for ESM
+    // it's not obvious at this point what a better abstraction would be for this, so leaving as-is
```

Comment:
Couple of options options:
1. We could just remove tsconfig initializer class and move it's content here.
2. Leave the logic that reads package json inside tsconfig intializer or pass `this.tsConfigInitializer.ensureInitialized(targetDir, packageJsonContent);`
3. Move this comment to tests. If the abstraction does not express correctness well the test might.

Reply:
1 is easy enough. It is still "generating project files" so I don't think it's wrong to move it to this class. Can always break it out later if it becomes unwieldy